improver/utilities/spatial.py

```diff
@@ -268,8 +280,16 @@ def calculate_difference(cube: Cube, coord_name: str) -> ndarray:
             Array after the differences have been calculated along the
             specified axis.
         """
-        diff_axis = cube.coord_dims(coord_name)[0]
-        diff_along_axis = np.diff(cube.data, axis=diff_axis)
+        diff_axis = cube.coord(name_or_coord=coord_name)
+        diff_axis_number = cube.coord_dims(coord_name)[0]
+        diff_along_axis = np.diff(cube.data, axis=diff_axis_number)
+        if self._axis_wraps_around_meridian(diff_axis, cube):
+            first_column = cube.data[:, :1]
```

Comment:
It would be clearer to have the index as 0 rather than :1.

As a separate issue it is would be better not to assume the order of the coordinates in a cube instead getting the index of the diff_axis from the cube.

Reply:
Have replaced the :1/:-1 with reshapes.

I'm not sure I understand the second point. Would be good to discuss.

Reply:
Have now fixed this. Does a single reshape if needed with [-1, 1] and takes the array axis to diff along from the cube dim-coord number rather than assuming the order of the coordinates on cube.data.